[ test_vlan_ping] [ test issue ] | 'str' object has no attribute 'decode'

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
test vlan/test_vlan_ping.py fail over python3 issue:
AttributeError: 'str' object has no attribute 'decode'

**Steps to reproduce the issue:**
1. run test_vlan_ping.py

**Describe the results you received:**

    @pytest.fixture(scope='module')
    def vlan_ping_setup(duthosts, rand_one_dut_hostname, ptfhost, nbrhosts, tbinfo):
        """
        Setup:      Collecting vm_host_info, ptfhost_info
        Teardown:   Removing all added ipv4 and ipv6 neighbors
        """
        vm_host_info = {}
    
        vm_name, vm_info = None, None
        topo_name = tbinfo["topo"]["name"]
        for nbr_name, nbr_info in list(nbrhosts.items()):
            if topo_name != "m0" or (topo_name == "m0" and "M1" in nbr_name):
                vm_name = nbr_name
                vm_info = nbr_info
                break
    
        py_assert(vm_name is not None, "Can't get neighbor vm")
>       vm_ip_with_prefix = (vm_info['conf']['interfaces']['Port-Channel1']['ipv4']).decode('utf-8')
E       AttributeError: 'str' object has no attribute 'decode'

**Describe the results you expected:**
test should not fail over test issue

**Additional information you deem important:**
**Output of `show version`:**
``` 
SONiC Software Version: SONiC.202211_RC8.2-c7dc07151_Internal
Distribution: Debian 11.6
Kernel: 5.10.0-18-2-amd64
Build commit: c7dc07151
Build date: Fri Mar 24 11:45:37 UTC 2023
Built by: sw-r2d2-bot@r-build-sonic-ci03-242

Platform: x86_64-mlnx_msn4600c-r0
HwSKU: Mellanox-SN4600C-C64
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2023X22082
Model Number: MSN4600-CS2FO
Hardware Revision: A1
Uptime: 13:35:27 up  7:51,  2 users,  load average: 1.42, 1.48, 1.45
Date: Mon 27 Mar 2023 13:35:28

Docker images:
REPOSITORY                                         TAG                               IMAGE ID       SIZE
docker-orchagent                                   202211_RC8.2-c7dc07151_Internal   52320ee407c4   530MB
docker-orchagent                                   latest                            52320ee407c4   530MB
docker-fpm-frr                                     202211_RC8.2-c7dc07151_Internal   fbd1a7cd6ad6   541MB
docker-fpm-frr                                     latest                            fbd1a7cd6ad6   541MB
docker-teamd                                       202211_RC8.2-c7dc07151_Internal   ac50920d456f   511MB
docker-teamd                                       latest                            ac50920d456f   511MB
docker-syncd-mlnx                                  202211_RC8.2-c7dc07151_Internal   0d52fb7ff023   930MB
docker-syncd-mlnx                                  latest                            0d52fb7ff023   930MB
docker-macsec                                      latest                            84361a51252d   513MB
docker-snmp                                        202211_RC8.2-c7dc07151_Internal   8e86fcf269da   540MB
docker-snmp                                        latest                            8e86fcf269da   540MB
docker-platform-monitor                            202211_RC8.2-c7dc07151_Internal   42741e515d4c   933MB
docker-platform-monitor                            latest                            42741e515d4c   933MB
docker-sonic-p4rt                                  202211_RC8.2-c7dc07151_Internal   8c77da5c8c72   577MB
docker-sonic-p4rt                                  latest                            8c77da5c8c72   577MB
docker-mux                                         202211_RC8.2-c7dc07151_Internal   d7f0ac34c936   543MB
docker-mux                                         latest                            d7f0ac34c936   543MB
docker-sonic-telemetry                             202211_RC8.2-c7dc07151_Internal   39608b5b2a71   793MB
docker-sonic-telemetry                             latest                            39608b5b2a71   793MB
docker-lldp                                        202211_RC8.2-c7dc07151_Internal   61e191dc3d58   537MB
docker-lldp                                        latest                            61e191dc3d58   537MB
docker-dhcp-relay                                  latest                            bba5d985f355   504MB
docker-database                                    202211_RC8.2-c7dc07151_Internal   fc887a4151bc   494MB
docker-database                                    latest                            fc887a4151bc   494MB
docker-eventd                                      202211_RC8.2-c7dc07151_Internal   53907e25504d   494MB
docker-eventd                                      latest                            53907e25504d   494MB
docker-router-advertiser                           202211_RC8.2-c7dc07151_Internal   356285e91054   494MB
docker-router-advertiser                           latest                            356285e91054   494MB
docker-sflow                                       202211_RC8.2-c7dc07151_Internal   780973343beb   479MB
docker-sflow                                       latest                            780973343beb   479MB
docker-nat                                         202211_RC8.2-c7dc07151_Internal   83454bb6925c   481MB
docker-nat                                         latest                            83454bb6925c   481MB
docker-sonic-mgmt-framework                        202211_RC8.2-c7dc07151_Internal   20cfc12ab3c5   612MB
docker-sonic-mgmt-framework                        latest                            20cfc12ab3c5   612MB
urm.nvidia.com/sw-nbu-sws-sonic-docker/doroce      1.1.0-202211-internal-5           078ba366e374   200MB
urm.nvidia.com/sw-nbu-sws-sonic-docker/sonic-wjh   1.4.0-202211-internal-5           b4f92852d9ca   323MB
```

--- a/tests/vlan/test_vlan_ping.py
+++ b/tests/vlan/test_vlan_ping.py
@@ -63,10 +63,7 @@ def vlan_ping_setup(duthosts, rand_one_dut_hostname, ptfhost, nbrhosts, tbinfo):
             break
 
     py_assert(vm_name is not None, "Can't get neighbor vm")
-    if six.PY2:
-        vm_ip_with_prefix = (vm_info['conf']['interfaces']['Port-Channel1']['ipv4']).decode('utf-8')
-    else:
-        vm_ip_with_prefix = vm_info['conf']['interfaces']['Port-Channel1']['ipv4']
+    vm_ip_with_prefix = six.ensure_text(vm_info['conf']['interfaces']['Port-Channel1']['ipv4'])
     output = vm_info['host'].command("ip addr show dev po1")
     vm_host_info["mac"] = output['stdout_lines'][1].split()[1]
     vm_ip_intf = ipaddress.IPv4Interface(vm_ip_with_prefix).ip

--- a/tests/vlan/test_vlan_ping.py
+++ b/tests/vlan/test_vlan_ping.py
@@ -3,6 +3,7 @@ import pytest
 import ipaddress
 import logging
 import ptf.testutils as testutils
+import six
 from tests.common.helpers.assertions import pytest_assert as py_assert
 
 logger = logging.getLogger(__name__)
@@ -62,7 +63,10 @@ def vlan_ping_setup(duthosts, rand_one_dut_hostname, ptfhost, nbrhosts, tbinfo):
             break
 
     py_assert(vm_name is not None, "Can't get neighbor vm")
-    vm_ip_with_prefix = (vm_info['conf']['interfaces']['Port-Channel1']['ipv4']).decode('utf-8')
+    if six.PY2:
+        vm_ip_with_prefix = (vm_info['conf']['interfaces']['Port-Channel1']['ipv4']).decode('utf-8')
+    else:
+        vm_ip_with_prefix = vm_info['conf']['interfaces']['Port-Channel1']['ipv4']
     output = vm_info['host'].command("ip addr show dev po1")
     vm_host_info["mac"] = output['stdout_lines'][1].split()[1]
     vm_ip_intf = ipaddress.IPv4Interface(vm_ip_with_prefix).ip

